internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -204,6 +215,61 @@ func getPackageMetadataFromZip(archivePath string) (packageMetadata, error) {
 	return getPackageMetadataFromZipReader(r, fileNamePrefix)
 }
 
+func skipFnFromZip(log *logger.Logger, r *zip.ReadCloser, detectedFlavor string, fileNamePrefix string, versionedHome string, registry map[string][]string) (install.SkipFn, error) {
+	if detectedFlavor == "" {
+		// no flavor don't skip anything
+		return func(relPath string) bool { return false }, nil
+	}
+
+	flavor, err := install.Flavor(detectedFlavor, "", registry)
+	if err != nil {
+		if errors.Is(err, install.ErrUnknownFlavor) {
+			// unknown flavor fallback to copy all
+			return func(relPath string) bool { return false }, nil
+		}
+		return nil, err
+	}
+	specsInFlavor := install.SpecsForFlavor(flavor) // ignoring error flavor exists, it was loaded before
+
+	// fix versionedHome
+	versionedHome = strings.ReplaceAll(versionedHome, "\\", "/")
+	log.Errorf("using %q", versionedHome)
+
+	readFile := func(specFilePath string) ([]byte, error) {
+		log.Errorf("reading %q", specFilePath)
```

Comment:
The errorf's are probably better as debugfs so we don't log every file in the upgrade every time, if you are planning to keep them.

Reply:
no these are just debug logs, i use error because it's easier to grep, removed